crates/holochain_types/src/app.rs

```diff
@@ -76,32 +75,54 @@ pub struct UpdateCoordinatorsPayload {
     pub source: CoordinatorSource,
 }
 
-/// The instructions on how to get the DNA to be registered
+/// The arguments to create a clone of an existing cell.
 #[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
 pub struct CreateCloneCellPayload {
-    /// Properties to override when installing this DNA
-    pub properties: Option<YamlProperties>,
-    /// The DNA to clone
-    pub dna_hash: DnaHash,
-    /// The Agent key with which to create this Cell
-    /// (TODO: should this be derived from the App?)
-    pub agent_key: AgentPubKey,
-    /// The App with which to associate the newly created Cell
-    pub installed_app_id: InstalledAppId,
-    /// The Role ID under which to create this clone
-    /// (needed to track cloning permissions and `clone_count`)
+    /// The app id that the DNA to clone belongs to
+    pub app_id: InstalledAppId,
+    /// The DNA's role id to clone
+    /// The Role id under which to create this clone
     pub role_id: AppRoleId,
-    /// Proof-of-membership, if required by this DNA
+    /// Properties to override for the new cell
+    pub properties: Option<YamlProperties>,
+    /// Optionally set the network seed for the new cell
+    pub network_seed: Option<NetworkSeed>,
+    /// Optionally set a proof of membership for the new cell
     pub membrane_proof: Option<MembraneProof>,
+    /// Optionally set a new origin time of the new cell
+    pub origin_time: Option<Timestamp>,
+    /// Optionally a name for the DNA clone
+    pub name: Option<String>,
 }
 
-impl CreateCloneCellPayload {
-    /// Get the CellId of the to-be-created clone cell
-    pub fn cell_id(&self) -> CellId {
-        CellId::new(self.dna_hash.clone(), self.agent_key.clone())
+/// Ways of identifying a clone cell.
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
+pub enum CloneCellId {
+    /// Clone id consisting of role id and clone index.
+    CloneId(CloneId),
+    /// Cell id consisting of DNA hash and agent pub key.
+    CellId(CellId),
+}
+
+impl fmt::Display for CloneCellId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            CloneCellId::CellId(id) => CellId::fmt(id, f),
+            CloneCellId::CloneId(id) => CloneId::fmt(id, f),
+        }
     }
 }
```

Comment:
FYI I think `derive_more::Display` would give something similar, not certain though.

Reply:
Thanks for pointing that out - that was new to me although I reasoned it must exist